WordPress/WordPressTest/NotificationUtility.swift

```diff
@@ -11,9 +11,7 @@ class NotificationUtility {
     }
 
     func tearDown() {
-        // Note: We'll force ContextManager override reset, since, for (unknown reasons) the ContextManager
-        // might be retained more than expected, and it may break other core data based tests.
-        contextManager.tearDown()
+        // Do nothing
     }
```

Comment:
Great suggestion! I think this class may warrant a slightly more clean up, since how it is used is a bit problematic, in theory, even though in practice it hasn't caused any issue.

This utility creates a `ContextManagerMock` instance internally, but the test cases which uses this utility may also create its own `ContextManagerMock` instance, using two databases in one test case doesn't seem like a good idea. I can clean this up by changing `NotificationUtility` initialiser to require an `CoreDataStack` argument, which can be passed in from test case classes, so that there is only one database during test run. Which mean both [`setUp`](https://github.com/wordpress-mobile/WordPress-iOS/blob/14aa10e1c70d3c6b24c78356d7536b6a747da806/WordPress/WordPressTest/NotificationUtility.swift#L9-L11) and `tearDown` methods in this class can be deleted. What do you think?

Reply:
> I can clean this up by changing NotificationUtility initialiser to require an CoreDataStack argument, which can be passed in from test case classes, so that there is only one database during test run. Which mean both [setUp](https://github.com/wordpress-mobile/WordPress-iOS/blob/14aa10e1c70d3c6b24c78356d7536b6a747da806/WordPress/WordPressTest/NotificationUtility.swift#L9-L11) and tearDown methods in this class can be deleted. What do you think?

That would be great! Dependency Injection 💪 💪 💪 It also fits nicely in the scope of the trial, streamlining how we interact with Core Data in the tests and moving towards a single source of truth. 👍